mlos_bench/mlos_bench/services/local/temp_dir_context.py

```diff
@@ -0,0 +1,57 @@
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+#
+"""
+Helper functions to work with temp files locally on the scheduler side.
+"""
+
+import logging
+from contextlib import nullcontext
+from tempfile import TemporaryDirectory
+from typing import Optional, Union
+
+from mlos_bench.services.base_service import Service
+
+_LOG = logging.getLogger(__name__)
+
+
+class TempDirContextService(Service):
```

Comment:
Might make sense to split out the `temp_dir_context` method from the `SupportsLocalExec` into `SupportsLocalTempDir` so that this can declare that it implements that and we can use it independently.  Otherwise, it's not entirely clear that this needs to be combined with other classes through inheritance in order to the full setup.

Reply:
Makes sense! This change was mostly to make pylint shut up about code duplication, but you're right, we should add it to the protocol

Reply:
mm, wait. local exec *always* depends on temp dir, and we never use temp dir outside of the local exec context, so we probably don't have to expose it to the user as an individual service. I'll make it an abstract class to make sure we never use it outside of local exec context, and write about it in the docstring. Let's talk and see if there is a more elegant solution to it, and do it in the subsequent PR.